share/completions/sops.fish

```diff
@@ -0,0 +1,147 @@
+complete -F -c sops -n __fish_is_first_arg -a help -d "shows a list of commands or help for one command"
+complete -F -c sops -n __fish_is_first_arg -a h -d "shows a list of commands or help for one command"
+
+# Primary flags without parameters
+complete -F -c sops -n __fish_is_first_token -s d -l decrypt -d "decrypt a file and output the result to stdout"
+complete -F -c sops -n __fish_is_first_token -s e -l encrypt -d "encrypt a file and output the result to stdout"
+complete -F -c sops -n __fish_is_first_token -s r -l rotate -d "generate a new data encryption key and reencrypt all values with the new key"
+complete -F -c sops -n __fish_is_first_token -s i -l in-place -d "write output back to the same file instead of stdout"
+complete -F -c sops -n __fish_is_first_token -s s -l show-master-keys -d "display master encryption keys in the file during editing"
+complete -F -c sops -n __fish_is_first_token -l extract -d "extract a specific key or branch from the input document. Decrypt mode only. Example: --extract '[\"somekey\"][0]'"
+complete -F -c sops -n __fish_is_first_token -l ignore-mac -d "ignore Message Authentication Code during decryption"
+complete -F -c sops -n __fish_is_first_token -l verbose -d "enable verbose logging output"
+complete -F -c sops -n __fish_is_first_token -l enable-local-keyservice -d "use local key service"
+complete -F -c sops -n __fish_is_first_token -s v -l version -d "print the version"
+
+# Primary flags with required parameters
+complete -x -c sops -n __fish_is_first_token -s k -l kms -d "comma separated list of KMS ARNs [\$SOPS_KMS_ARN]"
+complete -x -c sops -n __fish_is_first_token -s p -l pgp -d "comma separated list of PGP fingerprints [\$SOPS_PGP_FP]"
+complete -x -c sops -n __fish_is_first_token -s a -l age -d "comma separated list of age recipients [\$SOPS_AGE_RECIPIENTS]"
+complete -x -c sops -n __fish_is_first_token -l gcp-kms -d "comma separated list of GCP KMS resource IDs [\$SOPS_GCP_KMS_IDS]"
+complete -x -c sops -n __fish_is_first_token -l aws-profile -d "the AWS profile to use for requests to AWS"
+complete -x -c sops -n __fish_is_first_token -l azure-kv -d "comma separated list of Azure Key Vault URLs [\$SOPS_AZURE_KEYVAULT_URLS]"
+complete -x -c sops -n __fish_is_first_token -l hc-vault-transit -d "comma separated list of Vault's key URI (e.g. 'https://vault.example.org:8200/v1/transit/keys/dev') [\$SOPS_VAULT_URIS]"
+complete -x -c sops -n __fish_is_first_token -l input-type -d "currently json, yaml, dotenv and binary are supported. If not set, sop will use the file's extension to determine the type"
+complete -x -c sops -n __fish_is_first_token -l output-type -d "currently json, yaml, dotenv and binary are supported. If not set, sop will use the input file's extension to determine the output format"
+complete -x -c sops -n __fish_is_first_token -l add-gcp-kms -d "add the provided comma-separated list of GCP KMS key resource IDs to the list of master keys on the given file"
```

Comment:
`-x` means you're disabling file completion for this option but this option wants a file argument.
Please try `-r` intead. Same for some other options.

Reply:
These options wants a value instead of files though. I have `-l output` and `-l config` with `-r` because only these two options wants a file.

Reply:
Ah sorry, I think I commented on the wrong completion. Looks like there are surprisingly few options that take file arguments.